src/js/components/Tabs/Tabs.js

```diff
@@ -93,37 +180,145 @@ const Tabs = forwardRef(
       messages,
     })}`;
 
+    const onLeft = () => {
+      if (focusIndex !== 0) {
+        tabRefs[focusIndex - 1].current.focus();
+        setFocusIndex(focusIndex - 1);
+      }
+    };
+
+    const onRight = () => {
+      if (focusIndex < tabRefs.length - 1) {
+        tabRefs[focusIndex + 1].current.focus();
+        setFocusIndex(focusIndex + 1);
+      }
+    };
+
+    const onHome = () => {
+      if (focusIndex !== 0 && tabRefs.length > 0) tabRefs[0].current.focus();
+    };
+
+    const onEnd = () => {
+      if (focusIndex !== 0 && tabRefs.length > 0)
+        tabRefs[tabRefs.length - 1].current.focus();
+    };
+
     return (
-      <StyledTabs
-        ref={ref}
-        as={Box}
-        role="tablist"
-        flex={flex}
-        responsive={responsive}
-        {...rest}
-        background={theme.tabs.background}
-      >
-        <StyledTabsHeader
+      <Keyboard onLeft={onLeft} onRight={onRight} onHome={onHome} onEnd={onEnd}>
+        <StyledTabs
+          ref={ref}
           as={Box}
-          direction="row"
-          justify={justify}
-          alignSelf={alignControls}
-          flex={false}
-          wrap
-          background={theme.tabs.header.background}
-          gap={theme.tabs.gap}
-          {...tabsHeaderStyles}
-        >
-          {tabs}
-        </StyledTabsHeader>
-        <StyledTabPanel
+          role="tablist"
           flex={flex}
-          aria-label={tabContentTitle}
-          role="tabpanel"
+          responsive={responsive}
+          {...rest}
+          background={theme.tabs.background}
         >
-          {activeContent}
-        </StyledTabPanel>
-      </StyledTabs>
+          <Box direction={overflow && scroll ? 'row' : 'column'}>
+            {overflow && scroll && (
+              <Button
+                a11yTitle="Previous Tab"
+                icon={<Previous />}
+                disabled={disableLeftArrow}
```

Comment:
Could we avoid the state variables for this and update arrow state and just do:
```suggestion
                disabled={isVisible(tabRefs[0].current)}
```

Reply:
same for the "Next Tab" button

Reply:
`disabled={isVisible(tabRefs[0].current)}` will only check once if the tab is visible or not and then never update this value. I used a state variable so that disabled would update as we are moving through the tabs.